sympy/core/numbers.py

```diff
@@ -772,7 +772,7 @@ def __new__(cls, num, prec=None):
             num = num._mpf_
 
         if prec is None:
-            dps = 15
+            dps = mpmath.mp.dps
```

Comment:
My worry with this is that if someone messes with the precision of mpmath, it will change the behavior of SymPy. Right now, a lot of code in SymPy assumes that the default precision is 15. It should only use this precision if the float is an mpmath type. Perhaps it should just reuse [this function](https://github.com/sympy/sympy/blob/d12dd1a2c6b1f066e8d798fc06a162ce7ed4b3e2/sympy/core/numbers.py#L3568).